spec/integration/user_index_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.feature 'User Index Page', type: :feature do
+  let(:user) do
+    User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo',
+                bio: 'Teacher from Poland.', posts_counter: 0)
+  end
+
+  before do
+    user
+    visit users_path
+  end
+
+  scenario 'I can see the username, profile picture, and number of posts for each user' do
+    expect(page).to have_content(user.name)
+    expect(page).to have_selector('img[alt="User Profile Picture"]')
+    expect(page).to have_content(user.posts_counter)
+  end
+
```

Comment:
- [ ] Your test is only checking for one user. But this is the user index page. We should be checking for more than one user, at least. Kindly write tests that check for multiple users instead of just one.

- [ ] **[OPTIONAL]** I think it would be better to split this block into three with each scenario being checked.

Reply:
Suggested changes by the code reviewer has been implemented. Creating multiple users and looping through and checking all users. Screen shot attached below for your reference.
![image](https://github.com/baqar-abbas/blog-app-rails/assets/117091923/bae2a745-fd55-424f-a84f-b5fe04842bba)